docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,88 @@
+compilation-from-portable-pdb
+----------------------
+
+Compilation from portable PDBs today is not completely possible, but is desirable in order to help reconstruct a compilation from source provided via source link or embedded in a pdb. Motivation is derived from [roslyn 41395](https://github.com/dotnet/roslyn/issues/41395).
+
+This document is restricted to the following assumptions:
+
+1. Only builds using the `-deterministic` and published to the symbol server.
+2. Source generator and analyzer references are not needed for this task. They may be useful, but are out of scope for this document. 
+3. Any storage capacity used for PDBs and source should not impact this feature, such as compression algorithm. 
+4. Only Portable PDB files will be included for this spec. We can expand this feature past these once it is implemented and proven needed elsewhere. 
+
+This document will provide the expanded specification to the Portable PDB format. Any additions to that format will be ported to expand documentation provided in [dotnet-runtime](https://github.com/jnm2/dotnet-runtime/blob/26efe3467741fe2a85780b2d2cd18875af6ebd98/docs/design/specs/PortablePdb-Metadata.md#source-link-c-and-vb-compilers).
+
+## List of Expanded Information 
+
+#### Compiler Version
+
+This will be a full compiler version, and if possible the commit SHA for the build of the compiler. This will be used since compilations accros compiler versions are not guaranteed to be the same.
+
+#### Signing Key
+
+Only public keys will be included. The following command line switches trigger a signing key: 
+
+* `-keyfile`
+* `-keycontainer`
+
+Open question: Do we need to know if `-delaysign` or `-publicsign` was specified? 
+
+#### Source Encoding
+
+Encoding for source files that are not declared using BOM. Do we need this?
+
+#### Path Map
+
+Required to replicate any path 
```

Comment:
Actually, this needs to be omitted since the resulting PDB wouldn't be deterministic if we included it.
The paths stored in the PDB are going to be already normalized and we can just use them as they are when re-creating the compilation.

Reply:
Ok, so when we store it won't matter because we can assume it's already 1-1 mapping to make the right compilation? Makes sense. I'll remove